src/renderer/widgets/Staking/RewardDestination/model/destination-model.ts

```diff
@@ -0,0 +1,377 @@
+});
+
+const $api = combine(
+  {
+    apis: networkModel.$apis,
+    walletData: $walletData,
+  },
+  ({ apis, walletData }) => {
+    if (!walletData) return undefined;
+
+    return apis[walletData.chain.chainId];
+  },
+  { skipVoid: false },
+);
+
+const $transactions = combine(
+  {
+    api: $api,
+    walletData: $walletData,
+    pureTxs: $pureTxs,
+    txWrappers: $txWrappers,
+  },
+  ({ api, walletData, pureTxs, txWrappers }) => {
+    if (!api || !walletData) return undefined;
+
+    return pureTxs.map((tx) =>
+      transactionService.getWrappedTransaction({
+        api,
+        addressPrefix: walletData.chain.addressPrefix,
+        transaction: tx,
+        txWrappers,
+      }),
+    );
+  },
+  { skipVoid: false },
+);
+
+// Max validators
+
+sample({
+  clock: $api.updates,
+  source: $maxValidators,
+  filter: (maxValidators, api) => !maxValidators && Boolean(api),
+  fn: (_, api) => api!,
+  target: getMaxValidatorsFx,
+});
+
+sample({
+  clock: getMaxValidatorsFx.doneData,
+  target: $maxValidators,
+});
+
+// Transaction & Form
+
+sample({
+  clock: [flowStarted, formModel.output.formChanged],
+  source: {
+    walletData: $walletData,
+    wallets: walletModel.$wallets,
+    accounts: walletModel.$accounts,
+  },
+  filter: ({ walletData }) => Boolean(walletData),
+  fn: ({ walletData, wallets, accounts }, data) => {
+    const signatories = 'signatory' in data && data.signatory ? [data.signatory] : [];
+
+    return destinationUtils.getTxWrappers({
+      chain: walletData!.chain,
+      wallet: walletData!.wallet,
+      wallets,
+      account: walletData!.shards[0],
+      accounts,
+      signatories,
+    });
+  },
+  target: $txWrappers,
+});
+
+sample({
+  clock: $txWrappers.updates,
+  fn: (txWrappers) => {
+    const signatories = txWrappers.reduce<Account[][]>((acc, wrapper) => {
+      if (wrapper.kind === WrapperKind.MULTISIG) acc.push(wrapper.signatories);
+
+      return acc;
+    }, []);
+
+    const proxyWrapper = txWrappers.find(({ kind }) => kind === WrapperKind.PROXY) as ProxyTxWrapper;
+
+    return {
+      signatories,
+      proxyAccount: proxyWrapper?.proxyAccount || null,
+      isProxy: transactionService.hasProxy(txWrappers),
+      isMultisig: transactionService.hasMultisig(txWrappers),
+    };
+  },
+  target: formModel.events.txWrapperChanged,
+});
+
+sample({
+  clock: formModel.output.formChanged,
+  source: $rewardDestinationData,
+  filter: (rewardDestinationData, data) => Boolean(rewardDestinationData) || typeof data !== 'number',
+  fn: (rewardDestinationData, data) => {
+    if (typeof data === 'number') {
+      return {
+        ...(rewardDestinationData || ({} as RewardDestinationData)),
+        validators: Array(data).fill({ address: TEST_ADDRESS }),
+      };
+    }
+
+    if (Array.isArray(data)) {
+      return { ...rewardDestinationData!, validators: data! };
+    }
+
+    return { ...data! };
+  },
+  target: $rewardDestinationData,
+});
```

Comment:
This works for me in `BondExtra`
```suggestion
sample({
  clock: formModel.output.formChanged,
  target: $rewardDestinationData,
});
```